### PR TITLE
Initial mechanics for 80:20 fee split

### PR DESCRIPTION
**Pull request by gavofyork**
Also:
- Introduce extra functions for Imbalance manipulation;
- Store treasury pot in an account, letting total issuance account for it;
- Make parity-codec versions consistent.

TODO: 
- [x] Make `Author` trait work in order to provide funds to the block author. This is blocked on #2918. Might be worth getting this PR in in the meantime nonetheless. Also see #2941, #2929.

### CODE REVIEW

**Comment by niklasad1** [comment]
@gavofyork This PR contains a lot of unrelated changes probably because `parity codec version` was reverted. 

It is very hard to review, can you fix the unrelated changes?

**Comment by rphmeier** [comment]
> Make Author trait work in order to provide funds to the block author. This is blocked on #2918. 

You can block on #2941 instead and call  `<authorship::Module<T>>::author()` with a dummy `FindAuthor` implementation

File: node/runtime/src/lib.rs [hunk 1]
```diff
@@ -58,8 +60,8 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("node"),
 	impl_name: create_runtime_str!("substrate-node"),
 	authoring_version: 10,
-	spec_version: 98,
-	impl_version: 100,
+	spec_version: 99,
+	impl_version: 99,
```
[the remark this answers is not in the document]
Comment (gavofyork) [reply node/runtime/src/lib.rs]: yes, it can in this case, since `spec_version` incremented.

**Comment by gavofyork** [comment]
@niklasad1 should be easier now.

File: core/sr-primitives/src/traits.rs [hunk 2]
```diff
@@ -830,6 +830,110 @@ pub trait OpaqueKeys: Clone {
...
+	fn try_from_account(a: &AccountId) -> Option<Self>;
+}
+
+/// Provide a simply 4 byte identifier for a type.
+pub trait TypeId {
+	/// Simple 4 byte identifier.
+	const TYPE_ID: [u8; 4];
+}
+
+/// Format is b"para" ++ encode(parachain ID) ++ 00.... where 00... is indefinite trailing zeroes to fill AccountId.
```
Comment (rphmeier) [inline core/sr-primitives/src/traits.rs]:
```suggestion
/// Format is TYPE_ID ++ encode(parachain ID) ++ 00.... where 00... is indefinite trailing zeroes to fill AccountId.
```

File: srml/treasury/src/lib.rs [hunk 3]
```diff
@@ -70,18 +70,23 @@
...
+	Zero, EnsureOrigin, StaticLookup, CheckedSub, CheckedMul, AccountIdConversion
+}};
 use parity_codec::{Encode, Decode};
 use system::ensure_signed;
 
 type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
 type PositiveImbalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::PositiveImbalance;
 type NegativeImbalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::NegativeImbalance;
 
+const MODULE_ID: ModuleId = ModuleId(*b"py/trsry");
```
Comment (rphmeier) [inline srml/treasury/src/lib.rs]: what is `py`?

Comment (gavofyork) [reply srml/treasury/src/lib.rs]: *p*arit*y*

File: node/runtime/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DemiMarie-parity) [reply node/runtime/src/lib.rs]: Sorry about that!

**Pull request merged**